src/main/java/no/sikt/nva/EmbargoScraper.java

```diff
@@ -0,0 +1,103 @@
+        try {
+            var embargoes = convertStringToEmbargoObjects(contentFileAsString);
+            logger.info("FOLLOWING COLLECTION CONTAINS " + embargoes.size() + " EMBARGOES.");
+            return embargoes;
+        } catch (Exception e) {
+            logger.error(ERROR_OCCURRED_EXTRACTING_EMBARGOES);
+            return Collections.emptyList();
+        }
+    }
+
+    public static void checkForEmbargo(Record record, List<Embargo> embargoes) {
+        var handle = record.getId().toString();
+        if (containsHandle(embargoes, handle)) {
+            var potentialEmbargo = embargoes.stream()
+                                       .filter(embargo -> embargo.getHandle().equals(handle))
+                                       .findAny().orElse(null);
+            if (recordContainsEmbargoFile(record, Objects.requireNonNull(potentialEmbargo))) {
+                record.getContentBundle()
+                    .getContentFileByFilename(potentialEmbargo.getFilename())
+                    .setEmbargoDate(potentialEmbargo.getDate());
+            }
+        }
+    }
+
+    private static boolean containsHandle(List<Embargo> embargoes, String handle) {
+        return embargoes.stream().map(Embargo::getHandle).collect(Collectors.toList()).contains(handle);
+    }
+
+    private static List<Embargo> convertStringToEmbargoObjects(String contentFileAsString) {
+        var listOfStringObjects = Arrays.asList(
+            contentFileAsString.replaceAll(EMPTY_LINE_REGEX, StringUtils.EMPTY_STRING).split("\n"));
+        var embargoes = listOfStringObjects.stream()
+                            .map(EmbargoScraper::convertToEmbargo)
+                            .collect(Collectors.toList());
+        return filterOutSameEmbargoes(embargoes);
+    }
+
+    private static Embargo convertToEmbargo(String string) {
+        var embargoAsList = Arrays.asList(string.split(";"));
+        String handle = embargoAsList.get(0);
+        String filename = embargoAsList.get(1);
+        String date = embargoAsList.get(2);
+        return new Embargo(handle, filename, date);
+    }
+
+    private static List<Embargo> filterOutSameEmbargoes(List<Embargo> embargoes) {
+        List<Embargo> uniqueEmbargoesList = new ArrayList<>();
+        for (Embargo embargo : embargoes) {
+            var handlesList = uniqueEmbargoesList.stream()
+                                  .map(Embargo::getHandle)
+                                  .collect(Collectors.toList());
+            if (!handlesList.contains(embargo.getHandle()) && isNotTextFileOrThumbnail(embargo.getFilename())) {
+                uniqueEmbargoesList.add(embargo);
+            }
+        }
+        return uniqueEmbargoesList;
+    }
+
+    private static boolean isNotTextFileOrThumbnail(String filename) {
+        return !filename.contains(PDF_TXT)
+               && !filename.contains(PDF_JPG)
+               && filename.split("\\.").length == 2;
```

Comment:
filename.split.length == 2 høres også risky å filtrere bort. Better safe than sorry. Hva er det værste som kan skje hvis du legger til en fil for mye i embargo lista di og hva er det værste som kan skje hvis du har en for lite?

Reply:
Du har rett ja